libs/sdk-bindings/bindings-android/lib/src/main/kotlin/breez_sdk_notification/job/ConfirmTransaction.kt

```diff
@@ -22,38 +23,63 @@ data class AddressTxsConfirmedRequest(
     val address: String,
 )
 
-class RedeemSwapJob(
+class ConfirmTransactionJob(
     private val context: Context,
     private val fgService: SdkForegroundService,
     private val payload: String,
     private val logger: ServiceLogger,
     private var bitcoinAddress: String? = null,
 ) : Job {
     companion object {
-        private const val TAG = "RedeemSwapJob"
+        private const val TAG = "ConfirmTransactionJob"
     }
 
     override fun start(breezSDK: BlockingBreezServices) {
         try {
             val request = Json.decodeFromString(AddressTxsConfirmedRequest.serializer(), payload)
             this.bitcoinAddress = request.address
-            breezSDK.redeemSwap(request.address)
-            logger.log(TAG, "Found swap for ${request.address}", "INFO")
         } catch (e: Exception) {
-            logger.log(TAG, "Failed to manually redeem swap notification: ${e.message}", "WARN")
-            notifyFailure()
+            logger.log(TAG, "Failed to decode payload: ${e.message}", "WARN")
         }
+
+        this.bitcoinAddress?.let {address ->
```

Comment:
I think it would be nice to be able to recognize whether it is a swap or reverse swap without trying both.
If we pass a query param app_data in the webhook url it returns back in the notification payload.
See here where it is grapped from the query param: https://github.com/breez/notify/blob/main/http/router.go#L40
And here how it is populated into the payload: https://github.com/breez/notify/blob/main/breezsdk/init.go#L43
Perhaps we can use it here to identify if it is a swap or reverse swap?

Reply:
That's a good argument. I am ok then to leave this inexpensive check.